File: db-service/src/main/java/pl/piwowarczyk/dbservice/word/Word.java
package pl.piwowarczyk.dbservice.word;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Word {
    
    private Long wordNumber;
    private String polish;
    private List<String> english;
    
}

File: frontend/src/app/services/word/Word.ts

export class Word {
  wordNumber: number;
  polish: string;
  english: string[];

  constructor(values: Object = {}) {
    Object.assign(this, values);
  }
}

export interface WordCreationEntity {
  unitId: string;
  polish?: string;
  english?: string[];
}

export interface WordEditionEntity {
  unitId: string;
  wordNumber: number;
  polish?: string;
  english?: string[];
}